dxpy pins vulnerable cryptography

https://github.com/dnanexus/dx-toolkit/blob/9a9fdb5965a30e997228a30d43e88602002fa0bf/src/python/requirements.txt#L9

see https://github.com/advisories/GHSA-x4qr-2fvf-3mr5 cannot install patched versions

Upgrading cryptography is needed since a `TypeError` arises when using an older version of `cryptography` (<37).

This has been reported multiple times, [here](https://github.com/aws/aws-cli/issues/7325) is an example.

For my use care it prevents tests from passing locally when using `boto3`,

```

/Users/lleavitt/git/parsnips_project/parsnips-venv/lib/python3.9/site-packages/stor/__init__.py:48: in wrapper
    f = getattr(Path(path), name)
/Users/lleavitt/git/parsnips_project/parsnips-venv/lib/python3.9/site-packages/stor/base.py:52: in __new__
    elif utils.is_s3_path(path):
/Users/lleavitt/git/parsnips_project/parsnips-venv/lib/python3.9/site-packages/stor/utils.py:220: in is_s3_path
    from stor.s3 import S3Path
/Users/lleavitt/git/parsnips_project/parsnips-venv/lib/python3.9/site-packages/stor/s3.py:11: in <module>
    import boto3
/Users/lleavitt/git/parsnips_project/parsnips-venv/lib/python3.9/site-packages/boto3/__init__.py:17: in <module>
    from boto3.session import Session
/Users/lleavitt/git/parsnips_project/parsnips-venv/lib/python3.9/site-packages/boto3/session.py:17: in <module>
    import botocore.session
/Users/lleavitt/git/parsnips_project/parsnips-venv/lib/python3.9/site-packages/botocore/session.py:26: in <module>
    import botocore.client
/Users/lleavitt/git/parsnips_project/parsnips-venv/lib/python3.9/site-packages/botocore/client.py:15: in <module>
    from botocore import waiter, xform_name
/Users/lleavitt/git/parsnips_project/parsnips-venv/lib/python3.9/site-packages/botocore/waiter.py:18: in <module>
    from botocore.docs.docstring import WaiterDocstring
/Users/lleavitt/git/parsnips_project/parsnips-venv/lib/python3.9/site-packages/botocore/docs/__init__.py:15: in <module>
    from botocore.docs.service import ServiceDocumenter
/Users/lleavitt/git/parsnips_project/parsnips-venv/lib/python3.9/site-packages/botocore/docs/service.py:14: in <module>
    from botocore.docs.client import ClientDocumenter, ClientExceptionsDocumenter
/Users/lleavitt/git/parsnips_project/parsnips-venv/lib/python3.9/site-packages/botocore/docs/client.py:17: in <module>
    from botocore.docs.example import ResponseExampleDocumenter
/Users/lleavitt/git/parsnips_project/parsnips-venv/lib/python3.9/site-packages/botocore/docs/example.py:13: in <module>
    from botocore.docs.shape import ShapeDocumenter
/Users/lleavitt/git/parsnips_project/parsnips-venv/lib/python3.9/site-packages/botocore/docs/shape.py:19: in <module>
    from botocore.utils import is_json_value_header
/Users/lleavitt/git/parsnips_project/parsnips-venv/lib/python3.9/site-packages/botocore/utils.py:37: in <module>
    import botocore.httpsession
/Users/lleavitt/git/parsnips_project/parsnips-venv/lib/python3.9/site-packages/botocore/httpsession.py:46: in <module>
    from urllib3.contrib.pyopenssl import (
/Users/lleavitt/git/parsnips_project/parsnips-venv/lib/python3.9/site-packages/urllib3/contrib/pyopenssl.py:50: in <module>
    import OpenSSL.crypto
/Users/lleavitt/git/parsnips_project/parsnips-venv/lib/python3.9/site-packages/OpenSSL/__init__.py:8: in <module>
    from OpenSSL import SSL, crypto
/Users/lleavitt/git/parsnips_project/parsnips-venv/lib/python3.9/site-packages/OpenSSL/SSL.py:19: in <module>
    from OpenSSL.crypto import (
    
    E   TypeError: deprecated() got an unexpected keyword argument 'name'

```

Forcing a `cryptography` upgrade to 39 solves this issue, but i am unable to lock my environment due to the strict cryptography specs for dxpy.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -8,6 +8,7 @@ Categories for each release: Added, Changed, Deprecated, Removed, Fixed, Securit
 
 ### Changed
 
+* Bump allowed cryptography dxpy dependency version to 40.0.x
 * Tab completion in interactive executions now works with `libedit` bundled in MacOS and does not require externally installed GNU `readline`
 
 ### Fixed

--- a/src/python/requirements.txt
+++ b/src/python/requirements.txt
@@ -6,7 +6,7 @@ psutil>=5.9.3
 requests>=2.8.0,<=2.27.1
 cryptography==3.3.2; python_version < "3"
 cryptography==3.2.1; python_version > "3" and python_version < "3.6"
-cryptography>=3.4.2,<37; python_version > "3.5"
+cryptography>=3.4.2,<41; python_version > "3.5"
 gnureadline==8.0.0; sys_platform == "darwin" and python_version < "3.9" and platform_machine == "x86_64"
 pyreadline==2.1; sys_platform == "win32" and python_version < "3.5"
 pyreadline3==3.4.1; sys_platform == "win32" and python_version >= "3.5"

